docs/content/operations/including-extensions.md

```diff
@@ -13,22 +13,137 @@ Druid extensions can be specified in the `common.runtime.properties`. There are
+│   ├── twitter4j-core-3.0.3.jar
+│   └── twitter4j-stream-3.0.3.jar
+├── druid-kafka-eight
+│   ├── druid-kafka-eight-0.7.3.jar
+│   ├── jline-0.9.94.jar
+│   ├── jopt-simple-3.2.jar
+│   ├── kafka-clients-0.8.2.1.jar
+│   ├── kafka_2.10-0.8.2.1.jar
+│   ├── log4j-1.2.16.jar
+│   ├── lz4-1.3.0.jar
+│   ├── metrics-core-2.2.0.jar
+│   ├── netty-3.7.0.Final.jar
+│   ├── scala-library-2.10.4.jar
+│   ├── slf4j-log4j12-1.6.1.jar
+│   ├── snappy-java-1.1.1.6.jar
+│   ├── zkclient-0.3.jar
+│   └── zookeeper-3.4.6.jar
+└── mysql-metadata-storage
+    ├── jdbi-2.32.jar
+    ├── mysql-connector-java-5.1.34.jar
+    └── mysql-metadata-storage-0.8.0-rc1.jar
+```
 
-The Druid processes will then only load up jars from the local repository and will not try to go out onto the internet to find the maven dependencies.
+As you can see, there are three sub-directories ```druid-examples```, ```druid-kafka-eight``` and ```mysql-metadata-storage``` under ```druid_extensions```, each sub-directory denotes an extension that Druid might load.
+
+3) Prepare hadoop dependencies directories under root hadoop dependency directory. The file structure for hadoop dependencies is almost same as normal Druid extensions described above, except that there is an extra layer of folder that specifies which version of hadoop it contains. 
+
+Example:
+
+Suppose you specify `druid.extensions.hadoopDependenciesDir=/usr/local/druid/hadoop_druid_dependencies`, and you want to prepare both `hadoop-client` 2.3.0 and 2.4.0,
+
+Then under ```hadoop_druid_dependencies```, it should look like this,
+
+```
+hadoop_druid_dependencies/
+└── hadoop-client
+    ├── 2.3.0
+    │   ├── activation-1.1.jar
+    │   ├── avro-1.7.4.jar
+    │   ├── commons-beanutils-1.7.0.jar
+    │   ├── commons-beanutils-core-1.8.0.jar
+    │   ├── commons-cli-1.2.jar
+    │   ├── commons-codec-1.4.jar
+    ..... lots of jars
+    └── 2.4.0
+        ├── activation-1.1.jar
+        ├── avro-1.7.4.jar
+        ├── commons-beanutils-1.7.0.jar
+        ├── commons-beanutils-core-1.8.0.jar
+        ├── commons-cli-1.2.jar
+        ├── commons-codec-1.4.jar
+    ..... lots of jars
+```
+
+As you can see, under ```hadoop-client```, there are two sub-directories, each denotes a version of ```hadoop-client```.
+
+4) Tell Druid which extensions to load.  Now you have prepared your extension directories, if you want Druid to load a specific list of extensions under root extension directory, you need to specify `druid.extensions.toLoad`. Using the example above, if you want Druid to load ```druid-kafka-eight``` and ```mysql-metadata-storage```, then you can specify `druid.extensions.toLoad=["druid-kafka-eight", "mysql-metadata-storage"]`.
+
+If you specify `druid.extensions.toLoad=[]`, Druid won't load any extension from file system.
+
+If you don't specify `druid.extensions.toLoad`, Druid will load all the extensions under root extension directory.
+
+### How to prepare extensions directories using `tools pull-deps`?
```

Comment:
I think if we keep including all the base extensions in release tarballs (like @xvrl's newer 0.8.1 builds) then this point is not too important, since most people will be using the prebuilt extensions dir. Only people wanting to use their own extensions would have to decide how to get the jars.